cpp/include/cudf/quantiles.hpp

```diff
@@ -42,5 +42,36 @@ quantile(column_view const& input,
          interpolation interp = interpolation::LINEAR,
          order_info column_order = {});
 
+/* @brief Gathers the rows corrosponding to the desired quantiles.
+ *
+ * The indices used to gather rows are computed by interpolating between the
+ * index on either side of the desired quantile. Since some columns may be
+ * non-arithmetic, interpolation between rows is limited to non-arithmetic
```

Comment:
But what if I have a table of only numeric columns? Shouldn't I be able to use linear interpolation? Also, I think we need to be explicit about what is allowed. Which interpolations are non-arithmetic?

Reply:
Added comment to #4011 to address this in future PR.

https://github.com/rapidsai/cudf/issues/4011#issuecomment-582991395